src/main/java/com/wavesplatform/wavesj/Node.java

```diff
@@ -105,6 +123,24 @@ public long getBalance(String address, String assetId) throws IOException {
                 : send("/assets/balance/" + address + "/" + assetId, "balance").asLong();
     }
 
+
+    public Map<String, Long> getAssetDistribution(String assetId) throws IOException {
+        String path = String.format("/assets/%s/distribution", assetId);
+        HttpResponse r = exec(request(path));
+        return parse(r, ASSET_DISTRIBUTION);
+    }
+
+    public Map<String, Long> getAssetDistributionByHeight(String assetId, Integer height) throws IOException {
+        String path = String.format("/assets/%s/distribution/%d", assetId, height);
+        HttpResponse r = exec(request(path));
+        return parse(r, ASSET_DISTRIBUTION);
+    }
+
+    public List<AssetBalance> getAssetsBalance(String address) throws IOException {
```

Comment:
What about Map<String, Long> ? Is it any more convenient?

Reply:
This endpoint not returns a Map<String, Long> It returns json with asset balance and  asset details, with inner object issue tx info.